layouts/lists.tsx

```diff
@@ -36,15 +37,17 @@ const ListPageLayout = ({
       <div className="flex flex-col items-center pt-20 page-container grow md:pt-14">
         <div className="info-container md:px-16 container w-full min-h-[100px]">
           <Header classNames="px-0 md:px-0">
-            {list && (
-              <ListHeader
-                name={list.name}
-                numberOfContributors={numberOfContributors}
-                isPublic={list.is_public}
-                listId={list.id}
-                isOwner={isOwner}
-              />
-            )}
+            <ClientOnly>
+              {list && (
+                <ListHeader
+                  name={list.name}
+                  numberOfContributors={numberOfContributors}
+                  isPublic={list.is_public}
+                  listId={list.id}
+                  isOwner={isOwner}
+                />
+              )}
+            </ClientOnly>
```

Comment:
refactor (blocking): This needs to load server side. All the data is available in `getServerSideProps`. I removed `<ClientOnly />` and it works fine with no hydration errors.

```suggestion
            {list && (
              <ListHeader
                name={list.name}
                numberOfContributors={numberOfContributors}
                isPublic={list.is_public}
                listId={list.id}
                isOwner={isOwner}
              />
            )}
```

Reply:
It fails in this case as the `numberOfContributors` prop is gotten clientside. 
![image](https://github.com/open-sauced/app/assets/62995161/2999363d-f264-4798-8685-b498ce9094ed)

unless we want to fetch the contributors data in the server, this will keep throwing hydration errors

Reply:
We want to render as much markup server side for the list header. The only part of it that is coming from the client-side is the `numberOfContributors` prop, so we can just do this instead for this line in the list header component.

https://github.com/open-sauced/app/blob/e281e143feb776765e5a2fb57199366c0d9720a4/components/ListHeader/list-header.tsx#L53

```diff
          <div className="flex items-center gap-2 mt-4" id="contributorCount">
-            {numberOfContributors} Contributors
+            <ClientOnly>{numberOfContributors} Contributors</ClientOnly>
          </div>
```